.github/workflows/ci-android-emulator-tests.yml

```diff
@@ -10,6 +10,11 @@ on:
 permissions:
   contents: read
 
+# Cancel the workflow if a new one is triggered from the same PR, branch, or tag, except on main.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
```

Comment:
In some projects we found we had to add `github.ref || github.run_id` but can't remember why exactly now...

Reply:
Thanks for the input.

According to the [Github docs](https://docs.github.com/en/actions/using-jobs/using-concurrency#example-using-a-fallback-value) you should provide fallback with "||" if using a property that is not always defined.

And according to the [docs here](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context), `github.ref` should be defined at least for "push" and "pull_request" events (which we use for CI workflows), if I understand the docs correctly. It also says "This is only set if a branch or tag is available for the event type" but I don't see how that could happen for a push or pull request.

Maybe there are some other events where it doesn't work but it seems fine here. Or we'll fix it if anything breaks ;)

Reply:
> Maybe there are some other events where it doesn't work but it seems fine here. Or we'll fix it if anything breaks ;)

Yeah, it could've been for the occasional manually triggered run...